src/test/functional/queryServices.test/suggested-tags.integration.ts

```diff
@@ -193,5 +195,83 @@ describe('tags query tests - happy path', () => {
       expect(tags[1].name).to.equal('romance');
       expect(tags[2].name).to.equal('horror');
     });
+
+    it('should get time_updated from list table if item_tags time_updated is null', async () => {
+      await db('item_tags').truncate();
+      //add items in list tables so we can join on null time_updated
+      await db('list').insert([
+        {
+          user_id: 1,
+          item_id: 10,
+          resolved_id: 10,
+          given_url: 'http://def',
+          title: 'Sample Title',
+          time_added: date3,
+          time_updated: date3,
+          time_read: date3,
+          time_favorited: date3,
+          api_id: 'apiid',
+          status: 1,
+          favorite: 1,
+          api_id_updated: 'apiid',
+        },
+      ]);
+
+      await db('item_tags').insert([
+        {
+          user_id: 1,
+          item_id: 11,
+          tag: 'tag A',
+          status: 1,
+          time_added: date2,
+          time_updated: date2,
+          api_id: 'apiid',
+          api_id_updated: 'updated_api_id',
+        },
+        {
+          user_id: 1,
+          item_id: 12,
+          tag: 'tag B',
+          status: 1,
+          time_added: date1,
+          time_updated: date1,
+          api_id: 'apiid',
+          api_id_updated: 'updated_api_id',
+        },
+        {
+          user_id: 1,
+          item_id: 1,
+          tag: 'tag C',
+          status: 1,
+          time_added: date,
+          time_updated: date,
+          api_id: 'apiid',
+          api_id_updated: 'updated_api_id',
+        },
+        {
+          user_id: 1,
+          item_id: 10,
+          tag: 'null date',
+          status: 1,
+          time_added: null,
+          time_updated: null,
+          api_id: 'apiid',
+          api_id_updated: 'updated_api_id',
+        },
+      ]);
+
+      const res = await server.executeOperation({
+        query: GET_SUGGESTED_TAGS,
+        variables,
+      });
+
+      const tags = res.data?._entities[0].savedItemById.suggestedTags;
+      expect(tags.length).to.equal(3);
+      //tagC is not in top 3 as it has the lowest time_updated
+      //null date tag will get its time_updated from list table.
+      expect(tags[0].name).to.equal('tag A');
```

Comment:
Curious about this ordering -- shouldn't it 'null date' (date3), then 'tag A' (date2), then 'tag B' (date1), for descending order by date?

Reply:
ya, we do a `getTags` query with 3 matching tags.

i understood the intend of this query as to return top 3 recent tags (not in any particular order), so didn't change this logic

https://github.com/Pocket/list-api/blob/ea79076916fdbe79a74f3b9b51f1866afbe1b3ef/src/dataService/tagDataService.ts#L133

Reply:
Ok, seems like a requirements question to me! Mind bringing it to Slack and checking with Amber?